usecases/steg_asymptotic_2sensors/asymptotic_model_error.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from asymptotic_model import AsymptoticModel
+from bayes.parameters import ModelParameters
+from collections import OrderedDict
+
+
+class AsymptoticModelError:
+    """
+    Define a standard model error for a specific type of test (in this case a asymptotic model)
+
+    Note:
+        f(x) = mA + mB/x , where mA and mB are model parameters
+        to be calibrated and x are the sensor positions in the interval [x_start, x_end].
+        MatB and MatC have identical sensor positions and noise specs.
+        MatB = a + bx  
+        MatC = c + dx + ex²
+        and Mat_B_C = MatB / MatC
+        f(x): Mat_B_C = mA + mB/x
+
+    Attributes:
+        x_Mat_sensors (np.array): Array of positions x where Mat values are measured
+        data_MatB (np.array): Array of sensor readings at x_Mat_sensors
+        data_MatC (np.array): Array of sensor readings at x_Mat_sensors
+        data_Mat_B_C  : (np.array): Array of evaluated MatB / MatC data
+        asymptotic_model (AsymptoticModel): asymptotic forward model (mA+mB/x) , where 'mA' and 'mB' are model parameters

```

Comment:
I'm trying to understand your problem. If you have measurements for matB and matC, you can define individual model error and inference procedures to get the individual parameters - resulting either in values for a,b,c,d,e or in distributions (when using Bayesian inference). If you have measurements for MatB, MatC as well as Mat_B_C, you can solve the inference/optimization problem for mA and mB with the model error
```
me_Mat_B_C = data_Mat_B_C - (mA+mB/x)
```
So those are all individual problems that seem to be unrelated at a first glance. Now, what might be the problem that your are solving (correct me if I'm wrong) - you only have measurements for MatB and MatC, and you would like to estimate what are the values mA and mB. In this case you can still generate the model errors for MatB and MatC independently and solve for a, b and c, d, e (or all together, if you don't care about the computational effort), and then solve another problem for Mat_B_C, where the data is now not generated from experiments, but is the (potentially noisy) output of models B and C. The joint problem with all variables a,b,c,d,e,mA,mB would look like this
```
me_MatB = data_MatB - (a+b*x)
me_MatC = data_MatC - (c+d*x+e*x**2
me_Mat_B_C = (a+b*x)/(c+d*x+e*x**2) - (mA +mB/x)
```

Reply:
Jörg, thanks for your comments - and apologize for not replying earlier, I simply overlooked them..

You are correctly seeing my problem. Both variants are relevant to our models in general, but in this use case I have measured (virtual) data for MatB and MatC. Mat_B_C is not measured but a direct evaluation of the prior data series. (These two series might be represented by models with the parameters a,b and c,d,e.)
For now, I am not interested in models of MatB and MatC, resepctively, but only in the model for Mat_B_C with the parameters mA and mB (the "target property"). In principle, this asymptotic model can be generated from experimental data (data_MatB / data_MatC) or from separate models with a,b and c,d,e (thus, two options). As said, for now, only the last model error definition me_Mat_B_C is relevant.

Reply:
> 
> 
> Jörg, thanks for your comments - and apologize for not replying earlier, I simply overlooked them..
> 
> You are correctly seeing my problem. Both variants are relevant to our models in general, but in this use case I have measured (virtual) data for MatB and MatC. Mat_B_C is not measured but a direct evaluation of the prior data series. (These two series might be represented by models with the parameters a,b and c,d,e.)
> For now, I am not interested in models of MatB and MatC, resepctively, but only in the model for Mat_B_C with the parameters mA and mB (the "target property"). In principle, this asymptotic model can be generated from experimental data (data_MatB / data_MatC) or from separate models with a,b and c,d,e (thus, two options). As said, for now, only the model error definition me_Mat_B_C = data_MatB / data_MatC is relevant. (What I have not addressed: data_MatB and data_MatC, if measured, might have different data uncertainties, respectively...)

Reply:
So in your current implementation, I'm not sure why you need the parameters a, b, c, d, e at all. I would suggest to either remove those, or extend the model error such that you have your second option. In the first option, you essentially have data for Mat_B_C mat_B and mat_C and try to evaluate mA and mB in your combined model - this is essentially a fitting of a function Mat_BC = mA+mB/x with data pairs  Mat_B_C = mA/mB. So I don't understand your measurents, because you have now measured that twice (once Mat_B_C directly and once mA and mB that allow you to compute Mat_B_C). So in that scenario, I think Mat_B_C should not be a measurement and removed from the data.

Reply:
I personally think though that your second option is much more interesting. In this case, you have measurements for all three, and Mat_B_C is directly a function of the model output of matB and matC, thus the model error is in Mat_B_C is not computed using the measured data, but using the models matB and matC.